### PR TITLE
Replace completionUri with notification

### PR DESCRIPTION
**Pull request by lanchongyizu**
* add WaitNotificationJob job/baseTask/task/unit test
* keep the original WaitCompletionUri job
* remove completionUri from all OSes which includes schema/task/unit test

### CODE REVIEW

**Comment by rackhdmirror** [comment]
@RackHD/corecommitters please review.

**Comment by lanchongyizu** [comment]
@RackHD/corecommitters @zyoung51 @cgx027 @uppalk1

**Comment by lanchongyizu** [comment]
jenkins: depends on https://github.com/RackHD/on-taskgraph/pull/169 https://github.com/RackHD/on-http/pull/459

**Comment by lanchongyizu** [comment]
test this please

**Comment by rackhdmirror** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by iceiilin** [comment]
will need review from @cgx027

**Comment by rackhdmirror** [comment]
Build finished.

File: lib/task-data/tasks/wait-for-notification.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+'use strict';
+
+module.exports = {
+    friendlyName: 'Notification Trigger',
+    injectableName: 'Task.Wait.Notification',
+    implementsTask: 'Task.Base.Wait.Notification',
+    options: {},
+    properties: {}
+};
```
Comment (cgx027) [inline lib/task-data/tasks/wait-for-notification.js]: Will a schema needed for this task? @yyscamper .

**Comment by cgx027** [comment]
Looks good overall. :+1: after fix houndci issue and coverage drop.

Maybe consider add schema for newly added task, But it can also be a follow up action.

File: lib/jobs/wait-notification.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2016, EMC, Inc.
+'use strict';
+
+var di = require('di');
+
+module.exports = waitNotificationJobFactory;
+di.annotate(waitNotificationJobFactory, new di.Provide('Job.Wait.Notification'));
+    di.annotate(waitNotificationJobFactory,
```
Comment (yyscamper) [inline lib/jobs/wait-notification.js]: incorrect indent.

File: lib/task-data/tasks/wait-for-notification.js (hunk 1 above)
Comment (yyscamper) [inline lib/task-data/tasks/wait-for-notification.js]: Yes, now this is required to new task now.

File: spec/lib/jobs/wait-notification-spec.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            WaitNotificationJob.prototype, '_subscribeNodeNotification', function(_nodeId, callback) {
+                callback({
+                    nodeId: _nodeId
+                });
+            });
+    });
+
+    it("should run", function() {
+        var job = new WaitNotificationJob({}, {}, graphId);
+        job._run().then(function() {
```
Comment (yyscamper) [inline spec/lib/jobs/wait-notification-spec.js]: The promise must be returned

**Comment by changev** [comment]
test this please

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by yyscamper** [comment]
:+1:

**Comment by yyscamper** [comment]
The Jenkins errors is expected since it doesn't resolve the dependencies, but the "Jenkins-Dependency-PR" success, it proves this PR is correct after resolving dependencies. 

So it's OK to merge.

**Pull request merged**